robocrys/describe/structure.py

```diff
@@ -0,0 +1,639 @@
+                describe cation polyhedra instead of both cation and anion
+                polyhedra.
+            only_describe_bonds_once: Whether only describe bond lengths once.
+                For example, don't describe the bond lengths from Pb to I and
+                also from I to Pb.
+            latexify: Whether to latexify the description.
+            return_parts: Whether to return the individual parts of the
+                description as a :obj:`dict`, or the whole description as a
+                :obj:`str`.
+        """
+        self.distorted_tol = distorted_tol
+        self.describe_mineral = describe_mineral
+        self.describe_component_makeup = describe_component_makeup
+        self.describe_components = describe_components
+        self.describe_symmetry_labels = describe_symmetry_labels
+        self.describe_oxidation_state = describe_oxidation_states
+        self.describe_bond_lengths = describe_bond_lengths
+        self.bond_length_decimal_places = bond_length_decimal_places
+        self.cation_polyhedra_only = \
+            only_describe_cation_polyhedra_connectivity
+        self.only_describe_bonds_once = only_describe_bonds_once
+        self.latexify = latexify
+        self.return_parts = return_parts
+        self.angle_decimal_places = 0
+
+        self._da: DescriptionAdapter = None
+        self._seen_bonds: set = None
+
+    def describe(self, condensed_structure: Dict[str, Any]
+                 ) -> Union[str, Dict[str, str]]:
+        """Convert a condensed structure into a text description.
+
+        Args:
+            condensed_structure: The condensed structure data, formatted as
+                produced by :meth:`StructureCondenser.condense_structure`.
+
+        Returns:
+            A description of the structure. If
+            :attr:`StructureDescriber.return_parts` is ``False``, the
+            description will be returned as a :obj:`str`. If it is equal to
+            ``True``, the description will be returned as a :obj:`dict` with the
+            keys 'mineral', 'component_makeup' and 'components', each containing
+            the relevant part of the description.
+        """
+        self._da = DescriptionAdapter(condensed_structure)
+        self._seen_bonds = set()
+
+        description = {}
+
+        if self.describe_mineral:
+            description['mineral'] = self._get_mineral_description()
+
+        if self.describe_component_makeup:
+            description['component_makeup'] = (
+                self._get_component_makeup_summary())
+
+        if self.describe_components:
+            description['components'] = self._get_all_component_descriptions()
+
+        if not self.return_parts:
+            return " ".join(description[part] for part in
+                            ['mineral', 'component_makeup', 'components'])
+        else:
+            return description
+
+    def _get_mineral_description(self) -> str:
```

Comment:
any reason this needs to be private?

Reply:
(repeat this comment for most of the other methods in this class - is there a potential use for the user to call these methods themselves, individually)?